src/dispatch-payload.ts

```diff
@@ -54,31 +55,48 @@ export const DispatchPayload = async (
   // deno-lint-ignore no-explicit-any
   let resp: any = {};
 
-  switch (validEventType) {
-    case EventTypes.FUNCTION_EXECUTED:
-      resp = await RunFunction(
-        payload as InvocationPayload<FunctionInvocationBody>,
-        functionModule,
-      );
-      break;
-    case EventTypes.BLOCK_ACTIONS:
-      resp = await RunBlockAction(
-        payload as InvocationPayload<BlockActionInvocationBody>,
-        functionModule,
-      );
-      break;
-    case EventTypes.VIEW_SUBMISSION:
-      resp = await RunViewSubmission(
-        payload as InvocationPayload<ViewSubmissionInvocationBody>,
-        functionModule,
-      );
-      break;
-    case EventTypes.VIEW_CLOSED:
-      resp = await RunViewClosed(
-        payload as InvocationPayload<ViewClosedInvocationBody>,
-        functionModule,
-      );
-      break;
+  try {
+    switch (eventType) {
+      case EventTypes.FUNCTION_EXECUTED:
+        resp = await RunFunction(
+          payload as InvocationPayload<FunctionInvocationBody>,
+          functionModule,
+        );
+        break;
+      case EventTypes.BLOCK_ACTIONS:
+        resp = await RunBlockAction(
+          payload as InvocationPayload<BlockActionInvocationBody>,
+          functionModule,
+        );
+        break;
+      case EventTypes.VIEW_SUBMISSION:
+        resp = await RunViewSubmission(
+          payload as InvocationPayload<ViewSubmissionInvocationBody>,
+          functionModule,
+        );
+        break;
+      case EventTypes.VIEW_CLOSED:
+        resp = await RunViewClosed(
+          payload as InvocationPayload<ViewClosedInvocationBody>,
+          functionModule,
+        );
+        break;
+      default:
+        throw new UnhandledEventError(
+          `Received a ${eventType} payload but there is no matching handler for the ${functionCallbackId} function.`,
+        );
+    }
+  } catch (handlerError) {
+    if (handlerError.name === "UnhandledEventError") {
```

Comment:
Decided to check against the error `name` so it would be easier for places like our SDK to throw this same name of Error w/o having to depend on this runtime. There might be better ways to share code like this in the future, but this seemed like a flexible way to start.